src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkAppealAsRemittedUploadDecisionHandler.java

```diff
@@ -57,22 +71,46 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
                 .getCaseDetails()
                 .getCaseData();
 
-        Document mayBeDecisionDocument = asylumCase.read(UPLOAD_REMITTAL_DECISION_DOC, Document.class)
-            .orElseThrow(() -> new IllegalStateException("uploadRemittalDecisionDoc is not present"));
-
-        Document decisionDocumnent = new Document(mayBeDecisionDocument.getDocumentUrl(),
-            mayBeDecisionDocument.getDocumentBinaryUrl(), getRemittalDecisionFilename(asylumCase));
-
-        asylumCase.write(UPLOAD_REMITTAL_DECISION_DOC, decisionDocumnent);
+        getCollectionOfRemittalDocumentsForUi(asylumCase);
         asylumCase.write(APPEAL_REMITTED_DATE, dateProvider.now().toString());
         asylumCase.write(REHEARING_REASON, "Remitted");
         asylumCase.write(CASE_FLAG_SET_ASIDE_REHEARD_EXISTS, YES);
         addRemittedCaseNote(asylumCase);
 
-
         return new PreSubmitCallbackResponse<>(asylumCase);
     }
 
+    private void getCollectionOfRemittalDocumentsForUi(AsylumCase asylumCase) {
+        Optional<List<IdValue<RemittalDocument>>> mayBeExistingRemittalDocuments = asylumCase.read(REMITTAL_DOCUMENTS);
+        List<IdValue<RemittalDocument>> existingRemittalDocuments = mayBeExistingRemittalDocuments.orElse(Collections.emptyList());
+        Document decisionDocument = asylumCase.read(UPLOAD_REMITTAL_DECISION_DOC, Document.class)
+            .orElseThrow(() -> new IllegalStateException("uploadRemittalDecisionDoc is not present"));
+
+        Document renamedDecisionDocument = new Document(decisionDocument.getDocumentUrl(),
+            decisionDocument.getDocumentBinaryUrl(), getRemittalDecisionFilename(asylumCase));
+
+        //Replace the field by the renamed document.
+        asylumCase.write(UPLOAD_REMITTAL_DECISION_DOC, renamedDecisionDocument);
+
+        String index = String.valueOf(existingRemittalDocuments.size() + 1);
+        int indexForCollection = 1; //For the id as 11, 12
+        List<IdValue<DocumentWithMetadata>> otherDocuments = new ArrayList<>();
+        Optional<List<IdValue<DocumentWithDescription>>> mayBeOtherDocuments = asylumCase.read(UPLOAD_OTHER_REMITTAL_DOCS);
+        if (mayBeOtherDocuments.isPresent()) {
+            //Add metadata to the documents in collection
+            for (IdValue<DocumentWithDescription> otherDocument : mayBeOtherDocuments.get()) {
+                DocumentWithDescription documentWithDescription = otherDocument.getValue();
+                DocumentWithMetadata documentWithMetaData = new DocumentWithMetadata(documentWithDescription.getDocument().get(),
+                    documentWithDescription.getDescription().get(), LocalDate.now().toString(), DocumentTag.REMITTAL_DECISION);
+                otherDocuments.add(new IdValue<>(index + indexForCollection++, documentWithMetaData));
```

Comment:
Why are you using 2 indices here? Isn't the same this
                otherDocuments.add(new IdValue<>(index + indexForCollection++, documentWithMetaData));
                as
                otherDocuments.add(new IdValue<>(index++ , documentWithMetaData)); ?